services/question/src/controllers/questionController.ts

```diff
@@ -200,6 +200,13 @@ export const updateQuestion = async (req: Request, res: Response) => {
     }
 
     try {
+        const existingQuestion = await Question.findOne({
+            $or: [{ title: updates.title }, { description: updates.description }],
+        }).collation({ locale: 'en', strength: 2 });
+        if (existingQuestion) {
+            return handleBadRequest(res, 'A question with the same title or description already exists.');
+        }
```

Comment:
```suggestion
    try {
        const updatedQuestion = await Question.findOneAndUpdate({ id: parseInt(id, 10) }, updates, {
            new: true,
            runValidators: true,
        });
    
        if (!updatedQuestion) {
            return handleNotFound(res, 'Question not found');
        }
        handleSuccess(res, 200, 'Question updated successfully', updatedQuestion);
    } catch (error) {
        return handleBadRequest(res, 'A question with the same title or description already exists.');
    }
```
I believe this is handled by the try block wrapping `findOneAndUpdate()`? I think this can be used to simplify `addQuestion()` as well